internal/service/service_windows.go

```diff
@@ -145,7 +150,18 @@ func (sh windowsServiceHandler) shutdown(s chan<- svc.Status) error {
 	stopTimeoutCtx, stopCancel := context.WithTimeout(context.Background(), stopTimeout)
 	defer stopCancel()
 
-	err := sh.svc.Stop(stopTimeoutCtx)
+	stopErrChan := make(chan error, 1)
+	go func() {
+		stopErrChan <- sh.svc.Stop(stopTimeoutCtx)
+	}()
+
+	var err error
+	select {
+	case <-time.After(windowsServiceShutdownTimeout):
+		err = errors.New("the service failed to shut down in a timely manner")
```

Comment:
Should we os.Exit(1) here or is this ok to just exit as is? I'm ok with this just wanted to ask.

Reply:
What I like about not calling `os.Exit(1)` here is that it properly sets the service state to stopped and exits through the service handler using whatever communication exists there. 

I'm actually not 100% sure on the benefits of that, or if it's necessary, but that's my reasoning on doing it this way.

Reply:
Maybe redundant, but I'd be in favor of including how long we waited in this message